pkg/api/kptfile/v1/types.go

```diff
@@ -276,6 +276,24 @@ type Function struct {
 
 	// `ConfigMap` is a convenient way to specify a function config of kind ConfigMap.
 	ConfigMap map[string]string `yaml:"configMap,omitempty"`
+
+	// `Selectors` are used to specify resources on which the function should be executed
+	// if not specified, all resources are selected
+	Selectors []Selector `yaml:"selectors,omitempty"`
```

Comment:
We need to figure out how we want to introduce schema changes in kptfile. We have to look at all the places kptfile is round-tripped internally in kpt workflow and also across the fn orchestration in case of meta functions (kpt <---> function) and figure out a way to handle it. Some cases:

1. In a basic scenario: kpt (say beta.4) working with newer kptfile schema that supports resource selectors, roundtripping will cause loss of selector information.
2. newer version of kpt built with newer schema will work fine with round-tripping.
3. kpt metafunctions roundtripping kptfile where function's kptfile schema is not compatible with provided by the orchestrator.